genienlp/train.py

```diff
@@ -228,12 +222,35 @@ def should_log(iteration, log_every):
     return iteration % log_every == 0
 
 
-def do_validate(
-    iteration, args, model, numericalizer, val_iters, *, train_task, round_progress, task_progress, writer, logger
-):
+def validate(task, val_iter, model, args, num_print=10):
+    with torch.no_grad():
+        model.eval()
+        if isinstance(model, torch.nn.DataParallel):
+            # get rid of the DataParallel wrapper
+            model = model.module
+
+        generation_output = model.validate(val_iter, task)
```

Comment:
Can you also rename `generation_output` variable (here and everywhere else) and the `GenerationOutput` class to `validation_output` and `ValidationOutput`? This way, the term "generation" (generating text output) and "validation" (generating text output and then using gold answers to potentially calculate confidence scores or metrics) will have the same meaning everywhere.

Reply:
Sure